lib/getNextJobs.js

```diff
@@ -23,6 +45,18 @@ const getNextJobs = (workflowGraph, config) => {
         throw new Error('Must provide a PR number with "~pr" trigger');
     }
 
+    // Check if the current job is a stage setup and the startFrom job is a non setup stage job
+    if (
+        isStageSetup(config.trigger) &&
```

Comment:
All the below criteria must be met
1. trigger is a stage setup
2. event `startFrom` must be a stage job
3. event `startFrom` should not be stage setup
4. trigger and event `startFrom` must belong to the same stage

#4 is not currently checked.

pseudo code for discussion
```
if(isStageSetup(config.trigger)) {
 const startFromStageName = getStageName(workflowGraph, config.startFrom);
 const triggerStageName = ?;// Extract stage name using REGEX

  if(!isStageSetup(config.startFrom) && startFromStageName === triggerStageName) {
        jobs.add(config.startFrom);
        return Array.from(jobs);
  }
}
```